Content.Server/Abilities/Librarian/LibrarianPowersSystem.cs

```diff
@@ -0,0 +1,123 @@
+using Content.Server.Abilities.Mime;
+using Content.Server.Popups;
+using Content.Server.Spawners.Components;
+using Content.Shared.Actions;
+using Content.Shared.Alert;
+using Content.Shared.Roles;
+using Content.Shared.Speech.Muting;
+using Content.Shared.StatusEffect;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+
+namespace Content.Server.Abilities.Librarian
+{
+    public sealed class LibrarianPowersSystem : EntitySystem
+    {
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+        [Dependency] private readonly AlertsSystem _alertsSystem = default!;
+        [Dependency] private readonly StatusEffectsSystem _statusSystem = default!;
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+
+        private EntityCoordinates? libraryCoordinates;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<LibrarianPowersComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<LibrarianPowersComponent, ShushActionEvent>(OnShush);
+        }
+
+        private void OnComponentInit(EntityUid uid, LibrarianPowersComponent component, ComponentInit args)
+        {
+            //find location of library for power range
+            foreach (var spawner in _entityManager.EntityQuery<SpawnPointComponent>(false))
+            {
+                if (spawner.Job?.ID.Equals("Librarian") == true && TryComp<TransformComponent>(spawner.Owner, out var spawnerXform))
+                {
+                    libraryCoordinates = spawnerXform.Coordinates;
+                    break;
+                }
+            }
```

Comment:
Maps are dynamic and for all you know the library has just been rebuilt somewhere in maintenance by the crew.

Reply:
You're certainly right, although I wasn't quite sure how else to track it.